syntax = "proto2";
package antalk.apigw;
import "common.proto";
option java_package = "com.antalk.pb";
option cc_generic_services = true;

//登录请求
message LoginReq {
	optional string uid = 1;
	optional string username = 2;
	optional string password = 3;
	optional antalk.common.UserStatType online_status = 4;
	optional antalk.common.ClientType client_type = 5;
	optional string client_version = 6;
}

message LoginResp {
	optional string uid = 1;
	optional uint32 server_time = 2;
	optional antalk.common.ResultType result_code = 3;
	optional string result_string = 4;
	optional antalk.common.UserStatType online_status = 5;
	optional antalk.common.UserInfo user_info = 6;
}

//登出请求
message LogoutReq {
	optional string uid = 1;
	optional int64 time_stamp = 2;	
}

message LogoutResp {
}

//长轮询请求，用于server push
//register notify callback
message RegisterNoityReq {
	optional string uid = 1;
}

message Notify {
	optional string body = 1;
}

//Server表示由其他服务发送至网关
//踢人通知
message ServerKickoutNotifyReq {
	optional string uid = 1;
	optional antalk.common.KickoutReason reason = 2;
}

message ServerKickoutNotifyResp {
}

message SendMsgReq {
	optional antalk.common.MsgInfo msg_info = 1;
}

message SendMsgResp {
	optional int64 msg_id = 1;
	optional antalk.common.ResultType result_code = 2;
}

message NotifyMsgReq {
	optional antalk.common.MsgInfo msg_info = 1;	
}

message NotifyMsgResp {

}

service LoginService {
	rpc Login(LoginReq) returns (LoginResp);
	rpc Logout(LogoutReq) returns (LogoutResp);
	rpc RegisterNoity(RegisterNoityReq) returns (Notify);
	rpc ServerKickoutNotify(ServerKickoutNotifyReq) returns (ServerKickoutNotifyResp);
};

service MsgService {
	rpc SendMsg(SendMsgReq) returns (SendMsgResp);
	rpc NotifyMsg(NotifyMsgReq) returns (NotifyMsgResp);
}
